data-prepper-plugins/obfuscate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/obfuscation/action/OneWayHashAction.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.obfuscation.action;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.processor.obfuscation.ObfuscationProcessor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@DataPrepperPlugin(name = "hash", pluginType = ObfuscationAction.class, pluginConfigurationType = OneWayHashActionConfig.class)
+public class OneWayHashAction implements ObfuscationAction {
+    
+
+    private MessageDigest messageDigest = null;
+    private byte[] salt;
+    private String saltKey;
+    private static final Logger LOG = LoggerFactory.getLogger(ObfuscationProcessor.class);
+
+    @DataPrepperPluginConstructor
+    public OneWayHashAction(final OneWayHashActionConfig config) {
+        
+        this.salt = config.getSalt();
+        this.saltKey = config.getSaltKey();
+        try{
+            messageDigest = MessageDigest.getInstance(config.getHashFormat());            
```

Comment:
I'm wondering if we should allow the user to choose any available Java hash algorithm. Should we limit these to a subset?

Reply:
My suggestion is only include ones that are asked for/needed. I do not think that we should include weaker ones that are still supported in java for backwards compatibility. I have included SHA-512 as it was asked. Do you have suggestions on other frequently used/ asked-for algorithms?

Reply:
I see that you have an `@Pattern` on the `format` so that only `SHA-512` is allowed. This makes sense.